### PR TITLE
Fix loop-protect for empty lines in do-while loops

### LINKED ISSUE #5162: "SyntaxError: Unexpected token ;" due to having a ; after while (start <length)
Challenge [Bonfire: Chunky Monkey](http://www.freecodecamp.com/challenges/bonfire-chunky-monkey#?solution=function%20chunk(data%2C%20n)%20%7B%0A%20%20%2F%2F%20Break%20it%20up.%0A%20%20%20%20var%20result%20%3D%20%5B%5D%2C%20%0A%20%20%20%20%20%20%20%20i%2C%20%0A%20%20%20%20%20%20%20%20start%20%3D%20%200%2C%20%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20len%20%3D%20data.length%3B%0A%0A%20%20%20%20do%20%7B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20result.push(data.slice(start%2C%20start%2Bn))%3B%0A%20%20%20%20%20%20%20%20start%20%2B%3D%20n%3B%0A%0A%20%20%20%20%7D%20while%20(start%3C%20len)%3B%0A%0A%20%20%20%20return%20result%3B%0A%7D%0A%0Achunk(%5B%22a%22%2C%20%22b%22%2C%20%22c%22%2C%20%22d%22%5D%2C%202)%3B) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.80 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript
function chunk(data, n) {
  // Break it up.
    var result = [], 
        i, 
        start =  0, 
        
        len = data.length;

    do {
        
        result.push(data.slice(start, start+n));
        start += n;

    } while (start< len);

    return result;
}

chunk(["a", "b", "c", "d"], 2);
```

### LINKED ISSUE #9652: Semicolon on while statement causes challenge failure
#### Challenge Name

https://www.freecodecamp.com/challenges/smallest-common-multiple

#### Issue Description

Using the sample solution in the wiki article for this algorithm, for a do...while loop, if you include the required semicolon after the while, you get "SyntaxError: Unexpected token ;". If you take the semicolon out, you get a stop sign on the line saying that the semicolon is missed, but the challenge passes.

#### Browser Information
* Browser Name, Version: Chrome, 52.0.2743.60 beta-m
* Operating System: WIndows 10 Home
* Mobile, Desktop, or Tablet: Desktop

#### Your Code
```js
   do {
    quot = newArr[0] * loop * newArr[1];
    for (n = 2; n < newArr.length; n++) {
      if (quot % newArr[n] !== 0) {
        break;
      }
    }

    loop++;
  } while (n !== newArr.length);
```

#### Screenshot
scm

### PR DESCRIPTION
**Pull request by augmt**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX
#### Description

Closes #5162
Closes #9652

### CODE REVIEW

File: public/js/lib/loop-protect/loop-protect.js [hunk 1]
```diff
@@ -358,8 +358,10 @@ if (typeof DEBUG === 'undefined') { DEBUG = true; }
                 index++;
 
                 if (index === line.length && lineNum < (lines.length-1)) {
-                  lineNum++;
-                  line = lines[lineNum];
+                  do {
+                    lineNum++;
+                    line = lines[lineNum];
+                  } while (line.length === 0);
```
Comment (BerkeleyTrue) [inline public/js/lib/loop-protect/loop-protect.js]: Why a do while loop? It seems like this could be replaced by an if statement

**Comment by atjonathan** [comment]
@augmt, please update your commit message using `git commit --amend`,
so it doesn't have **"Closes #5162"**.

**Comment by raisedadead** [comment]
Added https://github.com/FreeCodeCamp/FreeCodeCamp/issues/9652

@augmt  

Thanks a lot for fixing this, particularly this loop-protect bug has been around since long.

There is a weird white space issue, although the fix works.
For instance if you open the issues mentioned in the OP, visit the challenges, copy paste the reported solution, the challenges do not pass.

However they pass once I refresh the page.
And the problem doesn't appear again.

Nevertheless that's a minor trouble. I think this pull request does the job that it aims to.
/cc @BerkeleyTrue I think this can be merged.

**Comment by BerkeleyTrue** [comment]
@raisedadead If it works please merge. Make sure it has no effect on other solutions as well.

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **LGTM.** :sparkles: :sparkles: 
@augmt Thanks a lot for fixing this!

**Pull request merged**